app/src/main/java/tech/pauly/findapet/discover/AnimalDetailsViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package tech.pauly.findapet.discover
+
+import android.databinding.ObservableBoolean
+import android.databinding.ObservableField
+import android.databinding.ObservableInt
+import tech.pauly.findapet.R
+import tech.pauly.findapet.data.models.Media
+import tech.pauly.findapet.data.models.Option
+import tech.pauly.findapet.data.models.PhotoSize
+import tech.pauly.findapet.shared.BaseViewModel
+import tech.pauly.findapet.shared.ResourceProvider
+import tech.pauly.findapet.shared.datastore.AnimalDetailsUseCase
+import tech.pauly.findapet.shared.datastore.TransientDataStore
+import javax.inject.Inject
+
+class AnimalDetailsViewModel @Inject
+internal constructor(dataStore: TransientDataStore,
+                     val detailsPagerAdapter: AnimalDetailsViewPagerAdapter,
+                     val imagesPagerAdapter: AnimalImagesPagerAdapter,
+                     private val resourceProvider: ResourceProvider) : BaseViewModel() {
+
+    var name = ObservableField("")
+    var age = ObservableInt(R.string.missing)
+    var breeds = ObservableField("")
+    var sex = ObservableInt(R.string.missing)
+    var size = ObservableInt(R.string.missing)
+    var options = ObservableField("")
+    var description = ObservableField("")
+    var descriptionVisibility = ObservableBoolean(false)
+    var optionsVisibility = ObservableBoolean(false)
+    var imagesPageLimit = ObservableInt(4)
+    var imagesCount = ObservableInt(0)
+    var currentImagePosition = ObservableInt(0)
+
+    init {
+        this.detailsPagerAdapter.setViewModel(this)
+
+        dataStore[AnimalDetailsUseCase::class.java]?.let {
```

Comment:
Now that these are all Kotlin classes, can we move to KClass to make this cleaner?

Reply:
Not doing this yet as Java classes still use this